compiler/rustc_codegen_ssa/src/debuginfo/type_names.rs

```diff
@@ -79,19 +92,21 @@ pub fn push_debuginfo_type_name<'tcx>(
                 hir::Mutability::Mut => output.push_str("mut "),
             }
 
-            push_debuginfo_type_name(tcx, inner_type, true, output, visited);
+            push_debuginfo_type_name(tcx, inner_type, qualified, output, visited);
 
             if cpp_like_names {
                 output.push('*');
             }
         }
         ty::Ref(_, inner_type, mutbl) => {
-            if !cpp_like_names {
+            if cpp_like_names {
+                output.push_str("ref ");
```

Comment:
We could go either way: currently `const` and `mut` are prefixes like this, rather than synthetic generic types. Personally I'd rather leave `ref` like this.